### PR TITLE
BFB-360: Raycast on eye height and center body height

### PR DESCRIPTION
**Pull request by m4rtint**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-360

**Description**
Raycast wall running on eye height

**Screenshots / GIFs**

https://youtu.be/p5-OzwXY2Js

### CODE REVIEW

File: Assets/_BForBoss/_Character/Scripts/PlayerWallRunBehaviour.cs [hunk 1]
```diff
@@ -245,10 +245,16 @@ public float CalculateWallSideRelativeToPlayer()
         
         private bool ShouldStartWallRun(out RaycastHit hit)
         {
-            var shouldStartWallRun =
-                ProcessRaycasts(_startWallRunDirections, out hit, ChildTransform, _parkourWallMask, _wallMaxDistance)
-                && IsClearOfGround()
-                && hit.collider != _lastWall;
+            var isRayCastingLastWallFromCenterBodyPosition = ProcessRaycasts(_startWallRunDirections, out hit,
```
Comment (Gursaanj) [inline Assets/_BForBoss/_Character/Scripts/PlayerWallRunBehaviour.cs:248]: up to you, but you can make a helper function to shorten the variable declaration which I think could be better for readability

```
private bool isRaycastingLastWall(Transform pivot)
{
   return ProcessRaycasts(_startWallRunDirections, out hit, pivot, _parkourWallMask, _wallMaxDistance) && hit.collider != lastWall;
}
```
then declare the variables as 
`var isRaycastingLastWallFromCenterBodyPosition = isRaycastingLastWall(ChildTransform);`
`var isRaycastingLastWallFromEyePosition = isRaycastingLastWall(ChildTransform);`

File: Assets/_BForBoss/_Character/Scripts/PlayerWallRunBehaviour.cs [hunk 2]
```diff
@@ -245,10 +245,16 @@ public float CalculateWallSideRelativeToPlayer()
...
+                                                                 ChildTransform, _parkourWallMask, _wallMaxDistance)
+                                                             && hit.collider != _lastWall;
+            var isRayCastingLastWallFromEyePosition = ProcessRaycasts(_startWallRunDirections, out hit,
```
Comment (Gursaanj) [inline Assets/_BForBoss/_Character/Scripts/PlayerWallRunBehaviour.cs:251]: For both variables, "RayCasting" should be "Raycasting" to follow convention

**Review by Gursaanj — COMMENTED** [review]
Does this change things in a way that having a gif/image could better show off what your changes accomplish

**Comment by m4rtint** [comment]
> 

https://youtu.be/p5-OzwXY2Js

File: Assets/_BForBoss/_Character/Scripts/PlayerWallRunBehaviour.cs (hunk 1 above)
Comment (m4rtint) [reply Assets/_BForBoss/_Character/Scripts/PlayerWallRunBehaviour.cs:248]: Normally writing a function just for that one line of code is a code smell. (case by case)

as long you already understand clearly what i'm trying to express there, I think it's fine. 
But if you still think it's better for readability then i can extract it.

**Review by Gursaanj — APPROVED** [review]

**Pull request merged**